packages/xod-arduino/platform/program.tpl.cpp

```diff
@@ -72,11 +72,30 @@ TransactionState g_transaction;
 
 #if defined(XOD_DEBUG) || defined(XOD_SIMULATION)
 namespace detail {
-void handleTweaks() {
-    if (XOD_DEBUG_SERIAL.available() > 0 && XOD_DEBUG_SERIAL.find("+XOD:", 5)) {
+void handleDebugProtocolMessages() {
+    {{#withTetheringInetNode nodes}}
+    bool tetheringInetNodeReceivingData = node_{{ id }}.output_INET->isReceiving();
```

Comment:
During the first transaction, this line executes **before** the `tethering-inet` node's `evaluate` function, where an instance of `TetheringInternet` is created. What's stored  in `output_INET` before that? Could that cause a crash?

Reply:
Oh, good catch!

Since the node state belongs to the program global state, I’m almost sure `output_INET` is set to `nullptr` before being initialized. @brusherru Please, verify. In any case, calling something through the null pointer is an error. I have no idea how it ever passes through it though.

Reply:
It uses static methods. Replaced with `TetheringInternet::isReceiving()` and `TetheringInternet::beginReceiving(...)` to get rid of pointer to the instance.
Thanks!